### PR TITLE
Refactor sync committee tests

### PR DESCRIPTION
**Pull request by ralexstokes**
This PR refactors the sync committee operation tests so that:

1. Rewards verification applies to all sync committee tests
    * there are multiple entries to running the spec processing handler and not all of them verified rewards (e.g. `run_successful_sync_committee_test` vs `run_sync_committee_processing`
    * fix by: consolidate test runners into helper code and target all into one "driver" function
2. Tests are generated with randomized variants of `SyncAggregate` 
    * we have randomized versions with other objects and part of this PR was created during an audit of the existing tests and new tests in Altair
    *  fix by: pulling out "randomized" tests into separate file, following the pattern in other parts of the test suite (e.g. `test_leak_basic.py` vs `test_leak_random.py`)
    * this portion is a refactoring to align w/ the rest of the code organization used throughout the repo

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate_random.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import random
```
Comment (djrtwo) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/test_process_sync_aggregate_random.py:1]: We might  consider putting both sync aggregate test files into a sub-directory `altair/block_processing/sync_aggregates/test_process_*.py`. I think this pattern would be more managable in the event of doing that type of sub-division in a phase with lots of tests. It's not critical here, but maybe marginally better... Thoughts?

**Review by djrtwo — APPROVED** [review]
looks good!

**Comment by ralexstokes** [comment]
NOTE: there are 27 spec tests generated from the `operations/sync_aggregate` files and I confirmed that 27 are still generated after this refactor (and extended the test gen diagnostics to include this info)

File: tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py [hunk 2]
```diff
@@ -502,150 +406,3 @@ def test_proposer_in_committee_with_participation(spec, state):
         else:
             state_transition_and_sign_block(spec, state, block)
     raise AssertionError("failed to find a proposer in the sync committee set; check test setup")
-
```
Comment (ralexstokes) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py:505]: the deletions from here and below are simply moved into the `..._random.py` file added

**Review by djrtwo — APPROVED** [review]

**Pull request merged**